src/System.Private.CoreLib/shared/System/Enum.cs

```diff
@@ -73,7 +74,7 @@ private string ValueToString()
                     return Unsafe.As<byte, UIntPtr>(ref data).ToString();
                 default:
                     Debug.Fail("Invalid primitive type");
-                    return null;
+                    return null!;
```

Comment:
I suggest just making this `return string.Empty;`.

Reply:
Offline I had suggested logging a bug and having us take this to a formal review and determine what we wanted to do for cases like this (both in the framework and as a general "Framework Design Guideline"). Did that bug get logged?

Namely, there are several internal methods where we have `default` cases to satisfy the compiler but that should never be hit themselves (in practice). For those, do we want to simply `return null!` (not all of these will have `string.Empty` or an equivalent), or do we want to do something a bit more like changing these to `throw` or `Environment.FailFast`?

Reply:
@stephentoub what if calling method throwing an exception with null return

@tannergooding created issue https://github.com/dotnet/coreclr/issues/23573

Reply:
>  do we want to do something a bit more like changing these to `throw` or `Environment.FailFast`

Both these are likely to generate slightly bigger code compared to just returning null. (The code size does not matter in this particular case.)

I think `return null!` is about as good as anything else. Empty string would be the next best option in this case.

Reply:
> Framework Design Guideline

"Framework Design Guideline" are about public surface. This is about coding style that we historically allowed a lot of flexibility for.

Reply:
> I think return null! is about as good as anything else.

@jkotas, how would you feel about removing the current default case, changing the byte case to be the default, and adding an assert to that branch that it's actually byte? That would seem to result in the best IL and would still assert.

Reply:
It is harder to read and understand as a general pattern.

I think the best general pattern is to throw exception, e.g. it would be `throw new InvalidOperationException(SR.InvalidOperation_UnknownEnumType);` in this case.